bin/extract_api_surface.py

```
@@ -0,0 +1,207 @@
+Usage:
+    python bin/extract_api_surface.py /path/to/meshtastic-package-dir
+    python bin/extract_api_surface.py /path/to/meshtastic-package-dir --class MeshInterface
+"""
+
+import ast
+import json
+import sys
+from pathlib import Path
+
+
+def _annotation_to_str(node) -> str:
```

Ruff (0.15.7)

[warning] Too many return statements (11 > 6)

(PLR0911)